src/rust/engine/process_execution/src/remote.rs

```diff
@@ -1395,53 +1531,49 @@ pub async fn ensure_action_uploaded(
   Ok(())
 }
 
-pub fn format_error(error: &bazel_protos::status::Status) -> String {
-  let error_code_enum = bazel_protos::code::Code::from_i32(error.get_code());
+pub fn format_error(error: &StatusProto) -> String {
+  let error_code_enum = Code::from_i32(error.code);
   let error_code = match error_code_enum {
-    Some(x) => format!("{:?}", x),
-    None => format!("{:?}", error.get_code()),
+    Code::Unknown => format!("{:?}", error.code),
+    x => format!("{:?}", x),
   };
-  format!("{}: {}", error_code, error.get_message())
+  format!("{}: {}", error_code, error.message)
 }
 
-fn rpcerror_to_status_or_string(
-  error: &grpcio::Error,
-) -> Result<bazel_protos::status::Status, String> {
-  match error {
-    grpcio::Error::RpcFailure(grpcio::RpcStatus {
-      status_proto_bytes: Some(status_proto_bytes),
-      ..
-    }) => {
-      let mut status_proto = bazel_protos::status::Status::new();
-      status_proto.merge_from_bytes(&status_proto_bytes).unwrap();
-      Ok(status_proto)
-    }
-    grpcio::Error::RpcFailure(grpcio::RpcStatus {
-      status, details, ..
-    }) => Err(format!(
-      "{:?}: {:?}",
-      status,
-      details
-        .as_ref()
-        .map(|s| s.as_str())
-        .unwrap_or_else(|| "[no message]")
-    )),
-    err => Err(format!("{:?}", err)),
-  }
-}
-
-pub(crate) fn rpcerror_to_string(error: grpcio::Error) -> String {
-  match error {
-    grpcio::Error::RpcFailure(status) => format!(
-      "{:?}: {:?}",
-      status.status,
-      status.details.unwrap_or_else(|| "[no message]".to_string())
-    ),
-    err => format!("{:?}", err),
-  }
+// fn rpcerror_to_status_or_string(
```

Comment:
This method became unnecessary because Tonic does not differentiate between a `RpcFailure` and a `RpcStatus` that is a failure code, while `grpcoio` did. This should be deleted.

Reply:
Amazing!